src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,225 @@
+	*/
+	void SendSetCoast();
+
+	/**
+	* @brief Add a set to the output buffer that will force the connected motor to spin at a given setpoint
+	* @param velocity_setpoint the raw 16-bit velocity command going to the motor
+	*/
+	void SendSetVelocitySetpoint(uint16_t velocity_setpoint);
+
+	/**
+	* @brief Returns the number of clients that we've initialized
+	* @return The value contained in _clients_in_use
+	*/
+	uint8_t GetNumberOfClients();
+
+	/**
+	* @brief Set all of the IQUART configuration init flags to true
+	*/
+	void MarkIquartConfigsForRefresh();
+
+	/**
+	* @brief Send a Get command to all of the parameters involved in IQUART configuration, and make sure the PX4 parameters and module values agree
+	*/
+	void UpdateIquartConfigParams();
+
+	/**
+	* @brief Until the timeout is reached, keep trying to update the PX4 parameters to match, as appropraite, the value on the module. This can
+	*	mean either setting the PX4 parameter to match the motor or vice versa
+	*/
+	void CoordinateIquartWithPx4Params(hrt_abstime timeout = 2_s);
+
+	/**
+	* @brief Gives access to the object ID currently being used
+	* @return The value stored in _object_id_now
+	*/
+	uint8_t GetObjectIdNow();
+
+	/**
+	* @brief When the target module ID parameter changes, we need to delete and remake all of our clients in order to make sure that we're talking to the
+	*        correct motor.
+	* @param new_object_id The new target module ID that we should use to instantiate our new clients
+	*/
+	void UpdateClientsToNewObjId(uint8_t new_object_id);
+private:
+	/**
+	* @brief Initialize all of the Vertiq Clients that we want to use
+	* @param object_id The object ID with which to initialize our clients
+	*/
+	void InitVertiqClients(uint8_t object_id);
+
+	/**
+	* @brief Initialize all of the combo_entry structs
+	*/
+	void InitComboEntries();
+
+	uint8_t _object_id_now;
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface *_serial_interface;
+
+	//IQUART Client configuration
+	IFCI _motor_interface;
+
+	//These exist whenever we're using IQUART
+	EntryWrapper<float, float> _velocity_max_entry;
+	EntryWrapper<float, float> _voltage_max_entry;
+	EntryWrapper<uint8_t, int32_t> _control_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _motor_direction_entry;
+	EntryWrapper<uint8_t, int32_t> _fc_direction_entry;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _throttle_cvi_entry;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _pulsing_voltage_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _x_cvi_entry;
+	EntryWrapper<uint8_t, int32_t> _y_cvi_entry;
+	EntryWrapper<float, float> _pulse_zero_angle_entry;
+	EntryWrapper<float, float> _pulse_velo_cutoff_entry;
+	EntryWrapper<float, float> _pulse_torque_offset_entry;
+	EntryWrapper<float, float> _pulse_volt_limit_entry;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
```

Comment:
use the configuration in order to set up our array properly. I only want to make one array, so just use macro magic.